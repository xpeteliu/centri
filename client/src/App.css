.mainContent {
  margin: 20px;
}

body {
  background-color: #E4F0D1;
}

.btn-primary, .btn-primary:hover, .btn-primary:active, .btn-primary:visited {
  background-color: #96BB88 !important;
  border-color: #96BB88 !important;
  color: #000000 !important;
}

.btn-secondary, .btn-secondary:hover, .btn-secondary:active, .btn-secondary:visited {
  background-color: #C1D9B8 !important;
  border-color: #C1D9B8 !important;
  color: #000000 !important;
}

.headerBar {
  background-color: #C1D9B8 !important;
}

.groupListItem {
  background-color: #FFFFFF;
  height: 80px;
  padding: 20px;
}

/* .App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-float infinite 3s ease-in-out;
  }
}

.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}

.App-link {
  color: rgb(112, 76, 182);
}

@keyframes App-logo-float {
  0% {
    transform: translateY(0);
  }
  50% {
    transform: translateY(10px);
  }
  100% {
    transform: translateY(0px);
  }
} */
