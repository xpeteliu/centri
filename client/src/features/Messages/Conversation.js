/* eslint react/prop-types: 0 */

import React from 'react';
import {
  Container, Card, Row, Col, Form, Button,
} from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

function Conversation(props) {
  const {
    messages, id, otherId, onSubmitMessage,
  } = props;

  messages.sort((a, b) => ((a.createdDate > b.createdDate) ? 1 : -1));

  const rows = [];
  messages.forEach((message) => {
    rows.push(<ConversationRow
      message={message}
      userId={id}
    />);
  });

  return (
    <Container className="d-flex flex-column justify-content-between">
      <Row>
        {rows}
      </Row>
      <Row>
        <Input
          handleSubmit={onSubmitMessage}
          userId={id}
          otherUserId={otherId}
        />
      </Row>
    </Container>
  );
}

function Input(props) {
  const { handleSubmit } = props;
  return (
    <Container className="p-3">
      <Form onSubmit={handleSubmit} controlId="formMessage">
        <Form.Group>
          <Form.Label>Chat</Form.Label>
          <Form.Control name="formMessageText" placeholder="..." />
        </Form.Group>
        <Button variant="primary" type="submit">
          Send
        </Button>
      </Form>
    </Container>
  );
}

function ConversationRow(props) {
  const { message, userId } = props;
  const { creatorId } = message;

  const alignLeft = creatorId !== userId;

  if (alignLeft) {
    return (
      <Row>
        <Col xs={6}>
          <Message
            message={message}
          />
        </Col>
        <Col />
      </Row>
    );
  }
  return (
    <Row>
      <Col />
      <Col xs={6}>
        <Message
          message={message}
        />
      </Col>
    </Row>
  );
}

function Message(props) {
  const { message } = props;
  const { content, createdDate } = message;
  const dateString = createdDate.toString();

  return (
    <Card>
      <Card.Body>
        <Card.Text>
          {content}
          <br />
          {dateString}
        </Card.Text>
      </Card.Body>
    </Card>
  );
}

export default Conversation;
