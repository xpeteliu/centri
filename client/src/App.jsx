import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route
} from 'react-router-dom';
import HomePage from './components/HomePage/HomePage';
import LoginPage from './components/LoginPage/LoginPage';
import './App.css';

const Home = () => (
  <HomePage />
);

const Login = () => (
  <LoginPage />
);

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <Route exact path="/" component={Home} />
          <Route path="/login" component={Login} />
        </div>
      </Router>
    );
  }
}

export default App;
