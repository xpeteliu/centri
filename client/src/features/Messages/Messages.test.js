import React from 'react';
import {
  act, render, screen, waitFor,
} from '@testing-library/react';
import { Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import { configureStore } from '@reduxjs/toolkit';
import App from '../../App';
import Conversation from './Conversation';
import userReducer from '../common/userSlice';
import modalReducer from '../common/MessageModal/modalSlice';

describe('Messages page', () => {
  const user1Id = '12345678901234567890abcd';
  const user2Id = '12345678901234567890efgh';

  const testGroup = {
    _id: '1234a',
    creatorId: user1Id,
    adminIds: [user1Id],
    memberIds: [user2Id],
  };

  const message1 = {
    _id: '1234a',
    content: 'message 1 text',
    senderId: user1Id,
    recipientId: user2Id,
    attachmentType: 'none',
  };

  const message2 = {
    _id: '1234b',
    content: 'message 2 text',
    senderId: user2Id,
    recipientId: user1Id,
    attachmentType: 'none',
  };

  const message3 = {
    _id: '1234c',
    content: 'message 3 text',
    senderId: user1Id,
    recipientId: user2Id,
    invitingGroupId: '1234group1234abc',
    attachmentType: 'none',
  };

  const message4 = {
    _id: '1234d',
    content: 'message 3 text',
    senderId: user1Id,
    recipientId: user2Id,
    invitingGroupId: '1234group1234abc',
    attachmentType: 'image',
    attachmentId: '1234a',
  };

  const message5 = {
    _id: '1234e',
    content: 'message 3 text',
    senderId: user1Id,
    recipientId: user2Id,
    invitingGroupId: '1234group1234abc',
    attachmentType: 'audio',
    attachmentId: '1234b',
  };

  const message6 = {
    _id: '1234f',
    content: 'message 3 text',
    senderId: user1Id,
    recipientId: user2Id,
    invitingGroupId: '1234group1234abc',
    attachmentType: 'video',
    attachmentId: '1234c',
  };

  const mockServer = setupServer(
    rest.get('/api/user/12345678901234567890abcd',
      (req, res, ctx) => res(ctx.json({
        _id: user1Id,
        username: 'testUser1',
      }))),
    rest.get('/api/user/12345678901234567890efgh',
      (req, res, ctx) => res(ctx.json({
        _id: user2Id,
        username: 'testUser2',
      }))),
    rest.post('/api/message/filter/paginate',
      (req, res, ctx) => res(ctx.json(
        [message1, message2, message3, message4, message5, message6]
      ))),
    rest.post('/api/group/filter/sort',
      (req, res, ctx) => res(ctx.json(
        [testGroup]
      ))),
  );

  beforeAll(() => { mockServer.listen(); });
  afterAll(() => { mockServer.close(); });

  test('should match snapshot', async () => {
    let container;
    const store = configureStore({
      reducer: {
        modal: modalReducer,
        user: userReducer,
      },
    });
    act(() => {
      ({ container } = render(
        <Provider store={store}>
          <Router history={createMemoryHistory({ initialEntries: ['/messages'] })}>
            <App />
          </Router>
        </Provider>,
      ));
    });
    expect(container)
      .toMatchSnapshot();
  });

  test('Click on convo', async () => {
    const store = configureStore({
      reducer: {
        modal: modalReducer,
        user: userReducer,
      },
      preloadedState: { user: { id: user1Id } },
    });
    act(() => {
      render(
        <Provider store={store}>
          <Router history={createMemoryHistory({ initialEntries: ['/messages'] })}>
            <App />
          </Router>
        </Provider>,
      );
    });
    const button = await screen.findByText(/testUser2/);
    act(() => {
      userEvent.click(button);
    });
  });

  test('Render convo', async () => {
    const store = configureStore({
      reducer: {
        modal: modalReducer,
        user: userReducer,
      },
      preloadedState: { user: { id: user1Id } },
    });
    act(() => {
      render(
        <Provider store={store}>
          <Router history={createMemoryHistory({ initialEntries: ['/messages'] })}>
            <Conversation
              messages={[
                message1, message2, message3, message4, message5, message6
              ]}
              id={user1Id}
              otherId={user2Id}
              otherName="testUser2"
              onSubmitMessage={null}
              onFileUpload={null}
              onAcceptInvite={null}
              onDeclineInvite={null}
            />
          </Router>
        </Provider>,
      );
    });
    await screen.findByText(/testUser2/);
  });
});