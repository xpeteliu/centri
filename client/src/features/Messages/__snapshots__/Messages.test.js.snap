// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages page click on convo 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="headerBar navbar navbar-expand-md navbar-light bg-light"
    >
      <div
        class="container"
      >
        <span
          class="navbar-brand"
          role="presentation"
        >
          Team20
        </span>
        <button
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              role="presentation"
            >
              Groups
            </a>
            <a
              class="nav-link"
              role="presentation"
            >
              Messages
            </a>
          </div>
          <div
            class="hstack gap-2"
          >
            <img
              class="rounded-circle"
              height="30"
              src="./photo.jpg"
              width="30"
            />
            <span
              class="navbar-text"
            />
          </div>
        </div>
      </div>
    </nav>
    <div
      class="App container"
    >
      <div
        class="w-100 container"
      >
        <h1>
          You have no messages!
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`Messages page should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="headerBar navbar navbar-expand-md navbar-light bg-light"
    >
      <div
        class="container"
      >
        <span
          class="navbar-brand"
          role="presentation"
        >
          Team20
        </span>
        <button
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
        >
          <div
            class="me-auto navbar-nav"
          />
          <div
            class="hstack gap-2"
          >
            <span
              class="navbar-text"
            />
          </div>
        </div>
      </div>
    </nav>
    <div
      class="App container"
    >
      <div
        class="w-100 container"
      >
        <h1>
          You have no messages!
        </h1>
      </div>
    </div>
  </div>
</div>
`;
